esp-hal/src/i2c/master/mod.rs

```diff
@@ -149,6 +149,86 @@ impl BusTimeout {
     }
 }
 
+#[cfg(not(esp32))]
+cfg_if::cfg_if! {
+    if #[cfg(esp32s2)] {
+        type FsmTimeoutType = u32;
+    } else {
+        type FsmTimeoutType = u8;
+    }
+}
+
+/// When the FSM remains unchanged for more than
+#[cfg_attr(esp32s2, doc = "the given amount of bus clock cycles")]
+#[cfg_attr(not(esp32), doc = "the 2^ the given amount of bus clock cycles")]
+/// a timeout will be triggered.
+///
+/// The default value is
+#[cfg_attr(esp32s2, doc = "0x100")]
+#[cfg_attr(not(esp32), doc = "0x10")]
```

Comment:
`not(esp32)` is true for `esp32s2`, too, so in the S2 docs both will be present I think

Reply:
🤦‍♂️